spec/bundler/source_list_spec.rb

```diff
@@ -301,18 +304,17 @@
       source_list.add_git_source("uri" => "git://first-git.org/path.git")
 
       expect(source_list.lock_sources).to eq [
+        Bundler::Source::Rubygems.new,
+        Bundler::Source::Rubygems.new("remotes" => "https://duplicate-rubygems.org"),
+        Bundler::Source::Rubygems.new("remotes" => "https://first-rubygems.org"),
+        Bundler::Source::Rubygems.new("remotes" => "https://second-rubygems.org"),
+        Bundler::Source::Rubygems.new("remotes" => "https://third-rubygems.org"),
         Bundler::Source::Git.new("uri" => "git://first-git.org/path.git"),
         Bundler::Source::Git.new("uri" => "git://second-git.org/path.git"),
         Bundler::Source::Git.new("uri" => "git://third-git.org/path.git"),
         Bundler::Source::Path.new("path" => "/first/path/to/gem"),
         Bundler::Source::Path.new("path" => "/second/path/to/gem"),
         Bundler::Source::Path.new("path" => "/third/path/to/gem"),
-        Bundler::Source::Rubygems.new("remotes" => [
-          "https://duplicate-rubygems.org",
-          "https://first-rubygems.org",
-          "https://second-rubygems.org",
-          "https://third-rubygems.org",
-        ]),
```

Comment:
Sounds good! I thought we had a spec for that type of case, but I'm not totally sure.